fw/pool.c

```diff
@@ -152,8 +151,14 @@ __tfw_pool_alloc(TfwPool *p, size_t n, bool align, bool *new_page)
 			: (void *)(c + 1);
 	}
 
-	a = (char *)TFW_POOL_CHUNK_END(p);
-	p->off += n;
+	if (align) {
+		a = (void *)TFW_POOL_ALIGN_SZ((unsigned long)TFW_POOL_CHUNK_END(p));
```

Comment:
The compiler also realizes that here you use aligned `p->off`, which you recompute on the next line, so it uses the aligned version of the offset. However, the code with two macros doesn't look simple. Maybe it's better to make `p->off = align ? TFW_POOL_ALIGN_SZ(p->off) : p->off` at the begin instead of `p_off` computation? (We don't care about the slow path.)

If you do the code adjustments, please make sure that the code is still well optimized and there are no double conditions and computations. This code is perfomace crucial - we call it on each TfwStr memory chunk.